parser.go

```diff
@@ -346,9 +355,34 @@ func (s *Session) readTemplateRecord(sl *slice) TemplateRecord {
 		tr.FieldSpecifiers[i] = f
 	}
 
+	tr.TemplateID = s.unaliasTemplateId(th.TemplateID, tr.FieldSpecifiers)
+
 	return tr
 }
 
+func (s *Session) unaliasTemplateId(templateId uint16, fieldSpecifiers []TemplateFieldSpecifier) uint16 {
+
+	s.mut.Lock()
+	defer s.mut.Unlock()
+
+	if id, ok := s.templateIdMap[templateId]; ok {
+		return id
+	}
+
+	var buffer bytes.Buffer
+	binary.Write(&buffer, binary.BigEndian, fieldSpecifiers)
+	hash := crc32.Checksum(buffer.Bytes(), crc32Table)
```

Comment:
Lets do at least a sha1 here to have some assurance that we don't have too many false positives in the deduplicating. Typically we don't process templates often enough for the slight CPU hit to matter.